src/bin/casr-python.rs

```diff
@@ -0,0 +1,329 @@
+            // Call casr-san
+            let mut python_cmd = Command::new("casr-san");
+            if let Some(report_path) = matches.value_of("output") {
+                python_cmd.args(["--output", report_path]);
+            } else {
+                python_cmd.args(["--stdout"]);
+            }
+
+            let output = python_cmd
+                .arg("--")
+                .args(&argv)
+                .stdout(Stdio::inherit())
+                .stderr(Stdio::inherit())
+                .output()
+                .expect("failed to start casr-san");
+
+            if output.status.success() {
+                return Ok(());
+            } else {
+                bail!("casr-san error");
+            }
+        }
+    } else if let Some(report_start) = python_stderr_list
+        .iter()
+        .position(|line| line.contains("Traceback "))
+    {
+        // Set python report in casr report.
+        if let Some(report_end) = python_stderr_list.iter().rposition(|s| !s.is_empty()) {
+            let report_end = report_end + 1;
+            if report_end > python_stderr_list.len() {
+                bail!("Corrupted output");
+            }
+            report.python_report = Vec::from(&python_stderr_list[report_start..report_end]);
+
+            let last = report
+                .python_report
+                .iter()
+                .skip(report_start)
+                .rposition(|s| !s.is_empty());
+            if last.is_none() {
+                bail!("Couldn't find traceback end in python report");
+            }
+            let last = last.unwrap();
+
+            let re = Regex::new(
+                r#"(File ".+", line [\d]+, in .+|\[Previous line repeated (\d+) more times\])"#,
+            )
+            .unwrap();
+            report.stacktrace = report.python_report[report_start..report_start + last]
+                .iter()
+                .rev()
+                .map(|s| s.trim().to_string())
+                .filter(|s| re.is_match(s))
+                .collect::<Vec<String>>();
+
+            // Get exception from python report.
+            if let Ok(exception) = python_exception(&report.python_report[report_start + last]) {
+                report.execution_class = exception;
+            }
+        } else {
+            bail!("Corrupted output");
+        }
+    } else {
+        // Call casr-san
+        let mut python_cmd = Command::new("casr-san");
+        if let Some(report_path) = matches.value_of("output") {
+            python_cmd.args(["--output", report_path]);
+        } else {
+            python_cmd.args(["--stdout"]);
+        }
+
+        let output = python_cmd
+            .arg("--")
+            .args(&argv)
+            .stdout(Stdio::inherit())
+            .stderr(Stdio::inherit())
+            .output()
+            .expect("failed to start casr-san");
+
+        if output.status.success() {
+            return Ok(());
+        } else {
+            bail!("casr-san error");
```

Comment:
Unclear Error. Casr-san started in some "stealth mode". Explain error: Couldn't start casr-san for...

Reply:
Done